crates/oxc_linter/src/rules/import/first.rs

```diff
@@ -0,0 +1,230 @@
+        .with_help("Move absolute import above relative import")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct First {
+    absolute_first: AbsoluteFirst,
+}
+
+#[derive(Debug, Default, Clone)]
+enum AbsoluteFirst {
+    AbsoluteFirst,
+    #[default]
+    DisableAbsoluteFirst,
+}
+
+impl AbsoluteFirst {
+    pub fn from(raw: &str) -> Self {
+        match raw {
+            "absolute-first" => Self::AbsoluteFirst,
+            _ => Self::DisableAbsoluteFirst,
+        }
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Forbids any non-import statements before imports except directives.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Notably, imports are hoisted, which means the imported modules will be evaluated
+    /// before any of the statements interspersed between them.
+    /// Keeping all imports together at the top of the file may prevent surprises
+    /// resulting from this part of the spec
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// import { x } from './foo';
+    /// export { x };
+    /// import { y } from './bar';
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// import { x } from './foo';
+    /// import { y } from './bar';
+    /// export { x, y }
+    /// ```
+    ///
+    /// ### Options
+    ///
+    /// with `"absolute-import"`:
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// import { x } from './foo';
+    /// import { y } from 'bar'
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// import { y } from 'bar';
+    /// import { x } from './foo'
+    /// ```
+    ///
+    First,
+    restriction,
+    pending  // TODO: fixer
+);
+
+/// <https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/first.md>
+impl Rule for First {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let obj = value.get(0);
+
+        Self {
+            absolute_first: obj
+                .and_then(serde_json::Value::as_str)
+                .map(AbsoluteFirst::from)
+                .unwrap_or_default(),
+        }
+    }
+
+    fn run_once(&self, ctx: &LintContext<'_>) {
+        let mut non_import_count = 0;
+        let mut any_relative = false;
+
+        let Some(root) = ctx.nodes().root_node() else {
+            return;
+        };
+        let AstKind::Program(program) = root.kind() else { unreachable!() };
+
+        for statement in &program.body {
+            match statement {
+                Statement::TSImportEqualsDeclaration(decl) => match decl.module_reference {
+                    TSModuleReference::ExternalModuleReference(ref mod_ref) => {
+                        if matches!(self.absolute_first, AbsoluteFirst::AbsoluteFirst) {
+                            if mod_ref.expression.value.as_str().starts_with("./") {
```

Comment:
1. we should probably extract this to a `is_relative_path` helper in this file at least.
2. should we support and check for `../` here as well?

Reply:
1. Ok I will try
2. No, it's just check for relative import only for case for absolute import setting